tools/clang/test/HLSLFileCheck/hlsl/types/union/union_cbuffer.hlsl

```diff
@@ -0,0 +1,16 @@
+// RUN: %dxc -E main -T ps_6_0 -enable-unions -HV 2021 %s | FileCheck %s
+
+// CHECK:define void @main
```

Comment:
It is not sufficient to test that `main` is defined in the generated IR. Tests should test that the output is correct as expected.

In this test case verifying the union member access IR is appropriate.

Other tests in this PR also need revision.

Reply:
Done